File: 1400 to 1500 difficulty/Mighty Friend (1406).java
//User
import java.util.*;
import java.lang.*;
import java.io.*;

class Codechef
{
    static int sumOfArray(ArrayList < Integer > arr)
    {
        int sum = 0;
        for (int i = 0; i < arr.size(); i++) {
            sum += arr.get(i);
        }
        return sum;
    }
    public static void main(String[] args) throws java.lang.Exception
    {
        Scanner sc = new Scanner(System.in);
        // your code goes here
        int T = sc.nextInt();
        while (T-- > 0) {
            int A = sc.nextInt();
            int swaps = sc.nextInt();
            ArrayList < Integer > Motu = new ArrayList < Integer > ();
            ArrayList < Integer > Tomu = new ArrayList < Integer > ();
            for (int i = 0; i < A; i++) {
                if (i % 2 == 0)
                    Motu.add(sc.nextInt());
                else
                    Tomu.add(sc.nextInt());
            }
            Collections.sort(Tomu);
            Collections.sort(Motu);
            Collections.reverse(Motu);
            int a = 0;swaps=Math.min(swaps,Tomu.size());
            while (swaps-- > 0 && (sumOfArray(Tomu) < sumOfArray(Motu)) ) {
                int t = Tomu.get(a);
                //System.out.println(t);
                //System.out.println("Soa:tomu: "+ sumOfArray(Tomu));
                Tomu.set(a, Motu.get(a));
                Motu.set(a, t);
                a++;
            }
            System.out.println(sumOfArray(Tomu) > sumOfArray(Motu) ? "YES" : "NO");
        }
    }
}

File: 1400 to 1500 difficulty/Mighty Friend (1406).cpp
#include <bits/stdc++.h>

using namespace std;
int sum(const vector < int > & myVector) {
    int result = 0;

    for (int num: myVector) {
        result += num;
    }

    return result;
}

pair < int, int > findMaxElementAndIndex(const int arr[], int size) {
    int maxElement = arr[0];
    int maxIndex = 0;

    for (int i = 1; i < size; ++i) {
        if (arr[i] > maxElement) {
            maxElement = arr[i];
            maxIndex = i;
        }
    }

    return make_pair(maxElement, maxIndex);
}

int main() {
    // your code goes here
    int t, n, swaps, temp, msum, tsum;
    cin >> t;
    while (t--) {
        msum = 0;
        tsum = 0;
        cin >> n >> swaps;
        vector < int > motu, tomu;
        for (int i = 0; i < n; i++) {
            cin >> temp;
            if (i % 2 == 0) {
                motu.push_back(temp);
                msum += temp;
            }
            else {
                tomu.push_back(temp);
                tsum += temp;
            }
        }
        if (tsum > msum) {
            cout<<"YES"<<endl;
        }
        else {
            sort(tomu.begin(), tomu.end());
            sort(motu.begin(), motu.end(), greater < int > ());
            int a = 0;
            while (swaps-- > 0 && (sum(tomu) < sum(motu))) {
                int t = tomu[a];
                tomu[a] = motu[a];
                motu[a] = t;
                a++;
            }
            sum(tomu)>sum(motu)?cout<<"YES"<<endl:cout<<"NO"<<endl;
        }
    }

}
